### DISCUSSION TITLE
Why no foreign key?

### DISCUSSION THREAD

**Discussion by glowkeeper** [directus/directus#8924] [Archive]
I have a table with a field `userid`, which looks like this in the directus app':

![m2o](https://user-images.githubusercontent.com/9264929/137781742-251a0bd7-f4da-4b73-b051-4f164f155592.png)

I was expecting to see that relationship modelled via foreign keys. However, when I do this in `mysql`:
```
show create table wallet;
```
It outputs this:
```
wallet | CREATE TABLE `wallet` (
  `id` int unsigned NOT NULL AUTO_INCREMENT,
  `user_created` char(36) DEFAULT NULL,
  `date_created` timestamp NULL DEFAULT NULL,
  `userid` varchar(36) NOT NULL,
  `address` varchar(255) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=9883 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
```
So `userid` doesn't appear to be set as a foreign key, which means it doesn't get indexed (I think). Am I missing something?

**Comment by rijkvanzanten** [comment]
What version of directus are you using?

**Reply by glowkeeper** [reply]
`9.0.0-rc.51`

**Reply by rijkvanzanten** [reply]
Ah that's why there's no foreign key then 🙂 Foreign key constraints where introduced in rc.70 👍🏻

**Reply by glowkeeper** [reply]
Okay. I've kinda been trying not to upgrade since I have >10k users and the system is kinda working. I guess I'm finding excuses to upgrade, though. So if I upgrade to the latest version, will I get foreign keys? And is the upgrade straightforward enough?

**Reply by rijkvanzanten** [reply]
I think all the features and improvements listed in https://github.com/directus/directus/blob/main/changelog.md should offer plenty of excuses to upgrade 😉 

> So if I upgrade to the latest version, will I get foreign keys?

This is correct.

> And is the upgrade straightforward enough?

It should be as simple as updating your Directus version, and running `directus database migrate:latest` to apply the DB changes required. That being said, if you're on a big production instance, I do definitely recommend spinning up a staging copy of that environment to try it out on first

**Reply by glowkeeper** [reply]
> spinning up a staging copy

Of course! Thanks...I'll form a plan...
